contracts/AtlasMineStakerUpgradeable.sol

```diff
@@ -873,40 +885,47 @@ contract AtlasMineStakerUpgradeable is
 
     /**
      * @dev Harvest rewards from the AtlasMine and send them back to
-     *      this contract.
+     *      this contract. Cannot harvest entire set of positions due to
+     *      gas limits, so positions need to be specified.
+     *
+     * @param depositIds            The deposits to harvest rewards for.
      *
      * @return earned               The amount of rewards earned for depositors, minus the fee.
      * @return feeEearned           The amount of fees earned for the contract operator.
      */
-    function _harvestMine() internal returns (uint256, uint256) {
+    function _harvestMine(uint256[] memory depositIds) internal returns (uint256, uint256) {
         uint256 preclaimBalance = magic.balanceOf(address(this));
 
-        try mine.harvestAll() {
-            uint256 postclaimBalance = magic.balanceOf(address(this));
+        for (uint256 i = 0; i < depositIds.length; i++) {
```

Comment:
theyre going to tell us to initialize `depositIds.length` outside of loop to save gas

Reply:
Good call, this is all over the place actually